synapse/storage/schema/main/delta/25/fts.py

```diff
@@ -37,7 +37,7 @@
 
 SQLITE_TABLE = (
     "CREATE VIRTUAL TABLE event_search"
-    " USING fts4 ( event_id, room_id, sender, key, value )"
+    " USING fts4 (tokenize=porter, event_id, room_id, sender, key, value )"
```

Comment:
I think this is desirable to avoid dropping and recreating this table later, but it might mean we need to update `full_schemas` which seems to be a cached copy of the materialized schema

Reply:
the problem is that if we do not update it, during creation of a new db will involve creating `event_search` and then dropping it and creating it again with `tokenize=porter`

Reply:
I don't think it makes sense to edit an old delta, as it means that there won't be consistency on what version 25 means regardless of when you installed Synapse (and note that delta 25 won't be run by new homeservers because new homeservers will start with the full snapshot at version 54).
The full schemas are point-in-time snapshots (not manually created), which would be equivalent to running all the deltas leading up to it.
We don't edit them once they've been released 'into the wild'.

Instead, the table should be dropped+recreated or altered in a new delta — yes it's slightly annoying for a new homeserver to have to do this 'create, drop, create' dance, but it takes negligible time. At a later date, a new full schema snapshot can be created so that new homeservers don't have to run all the historical deltas.
As for the background job: The old background job can be deleted and then recreated — this way you don't get two jobs queued up to do the same task.

Reply:
this delta (v25) won't get used by anyone creating a new db today, because we'll start from the full schema in `synapse/storage/schema/main/full_schemas/54`, and only apply the deltas from 55 onwards.

The only time delta 25 is used is if someone has an incredibly old installation (pre-#326) and attempts to upgrade that directly to a modern synapse version. It'd be *nice* if it worked, but there's certainly no need to make it efficient, and realistically we're not going to test upgrades from Synapse versions that old.